### DISCUSSION TITLE
Not all matches in `ispell-alternate-dictionary` are displayed.

### DISCUSSION THREAD

**Discussion by hongyi-zhao** [company-mode/company-mode#1380] [General]
I use the following setting with company:
```emacs
   ispell-alternate-dictionary (file-truename "~/Public/repo/github.com/hongyi-zhao/english-wordlist.git/american-english-exhaustive")
```
I test it as follows:
```shell
$ ug isomo ~/Public/repo/github.com/hongyi-zhao/english-wordlist.git/american-english-exhaustive
isomorph
isomorphic
isomorphically
isomorphism
isomorphism's
isomorphisms
isomorphous
isomorph's
isomorphs
isomorphy
unisomorphic
```

![image](https://user-images.githubusercontent.com/11155854/229413369-9f79e63c-7b40-419e-86f4-653c48fc623a.png)

As you can see, not all matches in `ispell-alternate-dictionary` are displayed. Any tips will be appreciated.

Regards,
Zhao

**Comment by dgutov** [comment]
Have you tried scrolling down in the completions list? The first two completions don't seem to come from ispell.

If the words are really not there, you can try stepping through `ispell-lookup-words` using edebug to see which dictionary actually gets used, and which output it provides.

**Reply by hongyi-zhao** [reply]
This is a complicated problem. See https://github.com/jcs-elpa/company-fuzzy/issues/37#issuecomment-1496275634 for more detailed discussion.

**Reply by dgutov** [reply]
Okay, I'll follow that discussion.

My recommendation to edebug `ispell-lookup-words` still stands, though.

**Reply by hongyi-zhao** [reply]
My suggestion is to take it step by step, starting with the most systematic clues.

**Reply by dgutov** [reply]
Either way, the other discussion also waits for your answers (to different questions, though).

**Reply by hongyi-zhao** [reply]
I've posted there my reply just now.

BTW, as for the `edebug` based method, what's the concrete steps to conduct it. I'm always confused on all sorts of Emacs debugging methods: Their logic and shortcut keys seem to be much more complicated than the usual operations.

**Reply by dgutov** [reply]
The second paragraph here: https://www.gnu.org/software/emacs/manual/html_node/elisp/Instrumenting.html
